lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -488,6 +494,100 @@ void DetectABI::interproceduralPropagation() {
+    auto &ABI = Summary.ABIResults;
+    auto &CFG = Summary.CFG;
+    auto &WrittenRegisters = Summary.WrittenRegisters;
+
+    const bool HasNotArguments = ABI.ArgumentsRegisters.size() == 0;
+    const bool HasNotReturns = ABI.FinalReturnValuesRegisters.size() == 0;
+    const bool IsSingleNode = CFG.size() == 1;
+
+    if (HasNotReturns and HasNotArguments and IsSingleNode) {
+      auto &Block = *CFG.begin();
+
+      if (Block.Successors().size() == 1) {
+        auto &Successor = *Block.Successors().begin();
+
+        // Select new prototype for wrapper function
+        if (auto Call = dyn_cast<efa::CallEdge>(Successor.get())) {
+          auto Prototype = getPrototype(*Binary, Entry, Block, *Call);
+
+
+          using abi::FunctionType::Layout;
+          // Get layout of wrapped function
+          Layout CalleeLayout = Layout::make(Prototype);
+
+          // Verify that wrapper function:
+          //  - don't write stack pointer
+          //  - don't write to callee arguments
+          //  - every store instruction writes to registers (not memory)
+          auto *BB = GCBI.getBlockAt(Block.ID().start());
+
+          llvm::GlobalVariable *StackPointer = nullptr;
+          {
+            const auto &Architecture = Binary->Architecture();
+            auto SP = model::Architecture::getStackPointer(Architecture);
+            llvm::StringRef StackPointerName = model::Register::getCSVName(SP);
+            StackPointer = M.getGlobalVariable(StackPointerName);
+          }
+
+          // Check if wrapper writes to Stack Pointer
+          const bool WriteSP = WrittenRegisters.count(StackPointer) > 0;
+
+          using std::ranges::count_if;
+          auto IsWritingToCalleeArg = [this,
+                                       &WrittenRegisters](auto &Argument) {
+            auto *CSV = M.getGlobalVariable(model::Register::getName(Argument));
+            return WrittenRegisters.count(CSV);
+          };
+          const auto &Arguments = CalleeLayout.argumentRegisters();
+          const bool WritesCalleeArgs = count_if(Arguments,
+                                                 IsWritingToCalleeArg);
+
+          auto IsWritingToMemory = [this](auto &Instruction) {
+            if (auto *StoreInst = dyn_cast<llvm::StoreInst>(&Instruction)) {
+              auto *Destination = StoreInst->getPointerOperand();
+
+              if (auto *GV = dyn_cast<llvm::GlobalVariable>(Destination)) {
+                auto &Architecture = Binary->Architecture();
+                auto Register = model::Register::fromCSVName(GV->getName(),
+                                                             Architecture);
+
+                // If destination is not register variable, it's write to memory.
+                if (GV->getName() == "pc") {
+                  return false;
+                }
+
+                return Register == model::Register::Invalid;
+              }
+            }
+
+            return false;
+          };
```

Comment:
Can you make a regular function? Do you really need to capture `this`?

If you do, consider making a method.

Reply:
I think we will still need to use some lambda to call method (`std::bind` is not generally recommended). In next change capturing `this` is removed, because most of this function is replaced with `isMemory` call. Should I squash _isMemory-commit_ to this?

Reply:
No need for a lambda if don't capure anything, we can just make a `static` function and call it a day.

> Should I squash isMemory-commit to this?

No fixup commits. All new code should not be amended by later commits.